ogr/ogrsf_frmts/hana/ogrhanalayer.cpp

```diff
@@ -765,41 +765,69 @@ OGRErr OGRHanaLayer::InitFeatureDefinition(const CPLString &schemaName,
 /*                         ReadGeometryExtent()                         */
 /************************************************************************/
 
-void OGRHanaLayer::ReadGeometryExtent(int geomField, OGREnvelope *extent)
+void OGRHanaLayer::ReadGeometryExtent(int geomField, OGREnvelope *extent,
+                                      int force, bool fastExtentMethod)
 {
     EnsureInitialized();
 
     OGRGeomFieldDefn *geomFieldDef = featureDefn_->GetGeomFieldDefn(geomField);
+    if (geomFieldDef == nullptr)
+        return;
+
     const char *clmName = geomFieldDef->GetNameRef();
     int srid = GetGeometryColumnSrid(geomField);
     CPLString sql;
-    if (dataSource_->IsSrsRoundEarth(srid))
+    if (!force && fastExtentMethod)
     {
-        CPLString quotedClmName = QuotedIdentifier(clmName);
-        bool hasSrsPlanarEquivalent = dataSource_->HasSrsPlanarEquivalent(srid);
-        CPLString geomColumn =
-            !hasSrsPlanarEquivalent
-                ? quotedClmName
-                : CPLString().Printf("%s.ST_SRID(%d)", quotedClmName.c_str(),
-                                     ToPlanarSRID(srid));
-        CPLString columns = CPLString().Printf(
-            "MIN(%s.ST_XMin()), MIN(%s.ST_YMin()), MAX(%s.ST_XMax()), "
-            "MAX(%s.ST_YMax())",
-            geomColumn.c_str(), geomColumn.c_str(), geomColumn.c_str(),
-            geomColumn.c_str());
-        sql = BuildQuery(rawQuery_.c_str(), columns.c_str());
+        auto names = dataSource_->FindSchemaAndTableNames(rawQuery_.c_str());
```

Comment:
Should we use a PreparedStatement? Then you could drop the literal wrapping

Reply:
I rewrote the function to do that, looks more clean now.